### PR TITLE
Turn a non shared volume into a shared volume when accessed via a RWX pvc

### PR DESCRIPTION
**Pull request by joshimoo**
Adds an api endpoint updateAccessMode to the volume api which is used by the csi driver to turn a non shared volume into a shared volume when accessed via a rwx pvc. it's only possible to call this api while the volume is detached.

https://github.com/longhorn/longhorn/issues/2048
https://github.com/longhorn/longhorn/issues/2071

### CODE REVIEW

**Comment by joshimoo** [comment]
manually tested turning a non shared volume into a shared volume via a RWX pvc.

**Review by shuo-wu — DISMISSED** [review]
In general LGTM except for missing the upgrade for existing volumes.

File: api/model.go [hunk 1]
```diff
@@ -889,7 +902,7 @@ func toVolumeResource(v *longhorn.Volume, ves []*longhorn.Engine, vrs []*longhor
 		RevisionCounterDisabled: v.Spec.RevisionCounterDisabled,
 		Ready:                   ready,
 
-		Share:         v.Spec.Share,
+		AccessMode:    v.Spec.AccessMode,
```
Comment (shuo-wu) [inline api/model.go:905]: If you decide not to update field `v.Spec.AccessMode` for existing volumes, you may still need to make sure this field exposed to UI is not empty.

**Comment by joshimoo** [comment]
added the upgrade part for existing volumes

**Comment by joshimoo** [comment]
Added a check at the manager level to ensure volume is detached before updating the access mode.

**Review by yasker — APPROVED** [review]

**Pull request merged**